src/core/Form/Checkbox/Checkbox.baseStyles.tsx

```diff
@@ -0,0 +1,175 @@
+import { element, font } from '../../theme/reset';
+
+/* stylelint-disable */
+const checkedStyles = ({ theme }: SuomifiThemeProp) => css`
+  &.fi-checkbox--checked .fi-checkbox_label {
+    &::before,
+    &::after {
+      border-color: ${theme.colors.highlightBase};
+    }
+  }
+`;
+
+const disabledStyles = ({ theme }: SuomifiThemeProp) => css`
+    &.fi-checkbox--disabled{    
+        & .fi-checkbox_label{
+            ${disabledCursor}
+    color: ${theme.colors.depthBase};
+        &::before, ::after{
+            background-color: ${theme.colors.depthLight3};
+            border-color: ${theme.colors.depthLight1};
+        }
+    }
+}
+`;
+
+const errorStyles = ({ theme }: SuomifiThemeProp) => css`
+  &.fi-checkbox--error {
+    & .fi-checkbox_label {
+      &::before {
+        border-color: ${theme.colors.alertBase};
+        border-width: 2px;
+      }
+      &::after {
+        border-color: ${theme.colors.alertBase};
+      }
+    }
+    & .fi-checkbox_hintText {
+      margin-bottom: ${theme.spacing.xxs};
+    }
+
+    & .fi-checkbox_status {
+      color: ${theme.colors.alertBase};
+    }
+  }
+`;
+
+const largeVariantStyles = ({ theme }: SuomifiThemeProp) => css`
+  &.fi-checkbox--large {
+    & .fi-checkbox_label {
+      padding-left: 40px;
+      line-height: 30px;
+      min-height: 30px;
+
+      &::before {
+        content: '';
+        position: absolute;
+        left: 0px;
+        top: 0px;
+        box-sizing: border-box;
+        height: 30px;
+        width: 30px;
+        color: ${theme.colors.depthBase};
+        border: 2px solid;
+      }
+      &::after {
+        position: absolute;
+        height: 8px;
+        width: 16px;
+        border-left: 4px solid;
+        border-bottom: 4px solid;
+        border-radius: 3px;
+        border-bottom-right-radius: ${theme.radius.basic};
+        border-top-left-radius: ${theme.radius.basic};
+        transform: rotate(-45deg);
+        left: 6px;
+        top: 6px;
+      }
+    }
+
+    & .fi-checkbox_hintText {
+      padding-left: 40px;
+    }
+  }
+`;
+
+export const baseStyles = withSuomifiTheme(
+  ({ theme }: TokensAndTheme) => css`
+    ${element({ theme })}
+    ${font({ theme })(
+      'bodyText',
+    )}
+    
+    /*TODO: use suomi.fi focus style */
+    &:focus-within {
+      & .fi-checkbox_label {
+        &::before {
+          box-shadow: 0 0 0 3px #ffbf47;
+          outline: 3px solid transparent; /* For Windows high contrast mode. */
+        }
+      }
+    }
+
+    & .fi-checkbox_input {
+      position: absolute;
+      opacity: 0;
+      z-index: -9999;
+      background-color: ${theme.colors.whiteBase};
+
+      & + .fi-checkbox_label::after {
+        content: none;
+      }
+      &:checked + .fi-checkbox_label::after {
+        content: '';
+      }
+    }
+    & .fi-checkbox_label {
+      position: relative;
+      display: inline-block;
+      padding-left: ${theme.spacing.l};
+      min-height: 30px;
+      min-width: 30px;
```

Comment:
These are still a bit speculative. The element needs to have a size to show the checkbox itself, but the values could be tweaked maybe.

Reply:
Are we able to say what should a min-width be? If we set something the I think the sizes should come from design. What is the display style of the element and how have we done this with other components? Maximum size could also be limited to window size.

Reply:
Actually, the max-width is yet to be set. Didn't see it set in other components either, so it's left for the user at this point.